app/services/manifest_token_json_converter.rb

```diff
@@ -0,0 +1,22 @@
+class ManifestTokenJsonConverter
```

Comment:
This isn't really a service. Why did you feel it needed to be factored out of the controller? It appears to just mangle incoming params for use by an actual service.